crates/ra_parser/src/grammar.rs

```diff
@@ -274,7 +274,7 @@ fn name(p: &mut Parser) {
 }
 
 fn name_ref(p: &mut Parser) {
-    if p.at(IDENT) {
+    if p.at(IDENT) || p.at(INT_NUMBER) {
```

Comment:
That would parse `0` as a valid name *anywhere*, which is probably not what we want.

I kinda agree that `INT_NUMBER` should be a child of `NameRef` though. Should we perhaps add a `allow_numeric_names: bool` parameter to this function?

Reply:
>Should we perhaps add a `allow_numeric_names: bool` parameter to this function

👌 let's do that!